### PR TITLE
Mediago Bid Adapter: initial adapter release

### PR DESCRIPTION
**Pull request by BaronJHYu**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [x] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

- test parameters for validating bids
```
{
  bidder: '<bidder name>',
  params: {
    // ...
  }
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer
- [ ] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/

### CODE REVIEW

**Comment by a-chatten** [comment]
Adapter docs: https://github.com/prebid/prebid.github.io/pull/3972

**Comment by BaronJHYu** [comment]
When will my commit be allowed to merge?

File: modules/mediagoBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+ * @param {Array}  validBidRequests an an array of bids
+ * @param {Object} bidderRequest  The master bidRequest object
+ * @return {Object}
+ */
+function getParam(validBidRequests, bidderRequest) {
+  const pubcid = utils.deepAccess(validBidRequests[0], 'crumbs.pubcid');
+  // console.log('wjh getParam', validBidRequests, bidderRequest);
+  let isMobile = isMobileAndTablet() ? 1 : 0;
+  let isTest = 0;
+  let auctionId = getProperty(bidderRequest, 'auctionId') || getRandomId();
```
Comment (Rothalack) [inline modules/mediagoBidAdapter.js:299]: I don't see a reason to have getRandomId. There should never be a scenario where a bidRequest does not have an auctionId here. Is there a reason you decided to create getRandomId for this?

File: modules/mediagoBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+    }
+
+    return bidResponses;
+  },
+
+  /**
+   * Register bidder specific code, which will execute if bidder timed out after an auction
+   * @param {data} Containing timeout specific data
+   */
+  onTimeout: function (data) {
```
Comment (Rothalack) [inline modules/mediagoBidAdapter.js:435]: I'm also unsure if onTimeout, onBidWon and onSetTargeting are necessary if you do not have anything happen within them. Did you have plans for these?

File: modules/mediagoBidAdapter.js (hunk 1 above)
Comment (BaronJHYu) [reply modules/mediagoBidAdapter.js]: All right,I will remove this code.

File: modules/mediagoBidAdapter.js (hunk 2 above)
Comment (BaronJHYu) [reply modules/mediagoBidAdapter.js]: got it

**Comment by BaronJHYu** [comment]
when i push code,CircleCi Safari 13.1.2 always Disconnected.How can I avoid this problem？

**Comment by BaronJHYu** [comment]
> when i push code,CircleCi Safari 13.1.2 always Disconnected.How can I avoid this problem？

already resubmit and resolved

**Review by Rothalack — APPROVED** [review]
Looks good, working in testing!

**Pull request merged**